src/historywork/ApplyBucketsWork.cpp

```diff
@@ -99,9 +85,18 @@ void
 ApplyBucketsWork::onStart()
 {
     auto& level = getBucketLevel(mLevel);
-    HistoryStateBucket& i = mApplyState.currentBuckets.at(mLevel);
+    HistoryStateBucket const& i = mApplyState.currentBuckets.at(mLevel);
     if (mApplying || i.snap != binToHex(level.getSnap()->getHash()))
     {
+        if (!mApplying)
```

Comment:
this will only be called when `i.snap != binToHex(level.getSnap()->getHash()` is true
what about moving that out of outside if so it becames
```
auto isCurrentSnapshot = i.snap != binToHex(level.getSnap()->getHash();
if (!isCurrentSnapshot) { ...}
if (mApplying || !isCurrentSnapshot) {...}
```